src/main/java/school/lemon/changerequest/java/introduction/pr1/Bob.java

```diff
@@ -1,10 +1,35 @@
 package school.lemon.changerequest.java.introduction.pr1;
 
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class Bob {
 
     public static String hey(String phrase) {
-        return "";
+        String phraseCheck=phrase.replaceAll(" ","");
+        if(phraseCheck.equals("")){return "Fine.";}
+        char question ='?';
+        char dot = '.';
+        char exclamation ='!';
+        char last= phrase.charAt(phrase.length()-1);
+        String anger= phrase.toUpperCase();
+        String notAnger= phrase.toLowerCase();
+        boolean notString=anger==notAnger;
+        if (last==question&&(notString||phrase!=anger))
+        {return "Yeap.";}
+
+        if (anger==phrase&&anger!=notAnger){return  "Chill out!";}
+
+        return "Whatever.";
```

Comment:
You've done lots of operations that could be avoided.
Try to solve task step-by-step, so that if you already know that it's question - don't do anything related to yelling.
Also, there is easier way to check if it's question or not - just use `endWith` `String` method.